code/modules/telesci/gps.dm

```diff
@@ -45,15 +45,25 @@ var/list/SPS_list = list()
 
 /obj/item/device/gps/emp_act(severity)
 	emped = TRUE
+	transmitting = FALSE
 	overlays -= image(icon = icon, icon_state = "working")
 	overlays += image(icon = icon, icon_state = "emp")
 	spawn(30 SECONDS)
-		emped = FALSE
 		overlays -= image(icon = icon, icon_state = "emp")
-		overlays += image(icon = icon, icon_state = "working")
+		emped = FALSE
 
 /obj/item/device/gps/attack_self(mob/user)
-	ui_interact(user)
+	if (emped)
+		return
+	else if (!transmitting)
+		switch(alert("Would you like to turn on the GPS?",,"Yes","No"))
```

Comment:
You should check again for the state of `transmitting` after you call `alert`, because it might have changed, because `alert` sleeps while waiting for the user. You also need to check the state of `emped` again, for the same reason. Also pass `user` as a parameter to alert, otherwise it will default to `usr`.

Reply:
I don't understand how alert sleeping affects the conditional and then by extension the state of emp and transmitting. I VV'd the variables while I had called alert but before I had entered a value and they were as they should be. I did find another issue though. people can remotely activate gps. how would I make a check for if the gps is in the user's hand?

Reply:
Yes, that's another thing to check for. You can use `Adjacent` to see if the GPS is still in reach. You should also check whether the user is `incapacitated`.
You can see the problem with alert for yourself: if you open the alert, then get EMP'd along with the GPS, then click "Yes" on the alert you'll have a perfectly functional GPS.

Reply:
well I tested it out. I emped while having the alert open. the emped gps opened the UI when I hit "YES", but everything was error and I couldn't interact with the UI. I took another GPS and checked the location of the first GPS, the one that got emped, and it gave me ERROR instead of it's coordinates, so I guess you could say it's working as intended. the only part that does bug out is that the emped visual overlay changes to a working one, even though the GPS is still emped.

Reply:
sure